### ISSUE TITLE
Stickers sent over MMS appear as an unopenable file on iOS devices

### ISSUE THREAD

**Issue by strugee** [signalapp/Signal-Android#9425]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

Stickers can be sent over MMS, but when sent to an iOS device, they appear as a file that can't be opened instead of as media.

### Steps to reproduce

1. Have a friend on iOS without Signal
2. Send the friend a sticker from a sticker pack (in my case, [this Ron Swanson sticker pack](https://signal.art/addstickers/#pack_id=81810744114022bac53fc51c9c50e982&pack_key=defaf5578bf7d322e2206a48a09ea2c4e2da0005815d68c52944dbbe22c4fbb0))

**Actual result:** the friend sees a file in the conversation which does not open when tapped
**Expected result:**  friend sees the sticker as an MMS message

### Screenshots

What the iOS recipient saw in my case:

![signal-attachment-2020-02-27-103131](https://user-images.githubusercontent.com/911174/75458939-a8a51880-594c-11ea-8ab3-52c2689eef1e.jpeg)

### Device info

**Device:** Pixel 3a XL
**Android version:** 10
**Signal version:** 4.56.1

### Link to debug log

This happened a couple days ago and I didn't think to capture a debug log. It doesn't seem worth bothering someone to capture one, but if it's needed I would be happy to do so.

**Comment by jeremy21212121** [comment]
[Stickers are encrypted](https://signal.org/blog/make-privacy-stick/). I don't understand how they could be received by a device that does not have signal. That being said, I'm not sure why the app lets you send stickers to non-signal users in the first place if it won't actually work.

I see nothing on the [relevant support page](https://support.signal.org/hc/en-us/articles/360031836512-Stickers) that suggests they can be received by non-signal users. On the contrary, that says users must have the latest version of signal to receive them.

NB - I have no affiliation with this project. I'm just a rando on the internet.

**Comment by strugee** [comment]
Yeah, I know they're encrypted. It looked to me like the selected sticker was being transcoded into a regular image. Maybe that's not what was happening though?

**Comment by greyson-signal** [comment]
The intention is that sticker sent over MMS should be sent as regular images. Looks like something isn't working right there, perhaps a carrier-specific problem. Who is your cellular carrier?

**Comment by strugee** [comment]
I'm on Ting's GSM network, which runs on T-Mobile's physical infrastructure.

**Comment by strugee** [comment]
She uses T-Mobile, apparently.

**Comment by strugee** [comment]
This has happened with a second friend who's using an iPhone 10 on AT&T. I'm now on Signal Beta 4.58.5.

**Comment by vyacheslav1** [comment]
Sticker is sent as image/webp attachment in MMS, which seems to be not supported on all platforms, even Android. For example, on Galaxy S8 with Android 9 such MMS is displayed in form of text: "image forbidden, file 6367739724921591785.webp deleted". I think, sticker should be converted to png before sending in MMS.

**Comment by strugee** [comment]
Since filing this report I have switched to Mint Mobile, which also runs (I believe exclusively?) on the T-Mobile network. Today I tested this with a friend running iOS (she had Signal installed, but presumably that shouldn't matter since on iOS texts only go to the system messaging app - and obviously I sent her a regular MMS message over the cell network for the test). My friend was able to successfully receive and display the big heart sticker from the Zozo the French Bulldog official sticker pack on iOS 14.4 running on an iPhone 11 and the T-Mobile network. I am currently running Android 11 on a Pixel 3a XL, Signal Beta 5.4.6.
